### ISSUE TITLE
Panic in propertiesRelatedTo for recursive variadic tuple constraint

### ISSUE THREAD

**Issue by HuzaifaAbdulRehman** [microsoft/typescript-go#4727]
## Stack trace

Tested with `@typescript/native-preview` version
`7.0.0-dev.20260707.2`.

```text
panic: runtime error: index out of range [-1] [recovered, repanicked]

github.com/microsoft/typescript-go/internal/checker.(*Relater).propertiesRelatedTo
    internal/checker/relater.go:4199

github.com/microsoft/typescript-go/internal/checker.(*Relater).structuredTypeRelatedToWorker
    internal/checker/relater.go:3864

github.com/microsoft/typescript-go/internal/checker.(*Relater).recursiveTypeRelatedTo
    internal/checker/relater.go:3135
```

## Steps to reproduce

Create `repro.ts`:

```ts
type Inner<A extends any[], A2 extends [...A, boolean]> = A2;

export type Wrapper<A extends any[]> = Inner<
  A,
  [...A, boolean, boolean, boolean, boolean]
>;
```

Run:

```sh
npx -y -p @typescript/native-preview@latest tsgo --noEmit repro.ts
```

The compiler panics in `propertiesRelatedTo`.

## Expected behavior

The compiler should report any applicable diagnostic without panicking.

## Related work

- Legacy issue: https://github.com/microsoft/TypeScript/issues/62974
- Unmerged JavaScript implementation: https://github.com/microsoft/TypeScript/pull/62983

The earlier PR passed its test runs but was automatically closed when ordinary
development moved from the JavaScript implementation to `typescript-go`.
